atlp-rwanda/rca-phantom-team2-bn
# (chore) setup: add continuous integration and quality assurance pipeline

#### What does this PR do?

- Setup Continuous Integration & Quality Assurance pipeline

#### Description of Task to be completed?

- Add the Configured Testing environment with Mocha, Chai, and Sinon to ensure that all functionalities are tested before deployment.
- Configure GitHub Actions as a continuous integration service used to build and test the app.
- Configure *** Pipeline on Render
- Add GitHub Actions  badge has been added to your repo
- Configure CodeClimate for code analysis and Maintainability checking.

#### How should this be manually tested?

- After cloning the repo you can run `npm install` to install the dependencies
- Checkout to branch `ch-Setup-Continuous-Integration-and-Quality-Assurance-pipeline`
- In order to start the project you can run `npm start`
- While checking the GitHub actions status you can check its badge in the `README.md`
- While checking the Code climate status you can check its badge in the `README.md`
- If you wish to access the project's URL on production, you can use this [Link](https://phantom-team2.onrender.com)

#### Any background context you want to provide?

None

#### What are the relevant pivotal tracker/Trello stories?

[Trello link](https://trello.com/c/CBg8otOV/18-setup-continuous-integration-quality-assurance-pipeline)

#### Screenshots (if appropriate)

#### Questions:
